internal/blobmanager/s3/provider.go

```diff
@@ -0,0 +1,121 @@
+//
+// Copyright 2023 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package s3
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	backend "github.com/chainloop-dev/chainloop/internal/blobmanager"
+	"github.com/chainloop-dev/chainloop/internal/credentials"
+)
+
+type BackendProvider struct {
+	cReader credentials.Reader
+}
+
+var _ backend.Provider = (*BackendProvider)(nil)
+
+func NewBackendProvider(cReader credentials.Reader) *BackendProvider {
+	return &BackendProvider{cReader: cReader}
+}
+
+const ProviderID = "AWS-S3"
+
+func (p *BackendProvider) ID() string {
+	return ProviderID
+}
+
+func (p *BackendProvider) FromCredentials(ctx context.Context, secretName string) (backend.UploaderDownloader, error) {
```

Comment:
and this implements the other interface we use to connect to CAS and control-plane. Which initializes + validates a backend from some stored creds